src/containers/DeleteConfirmation.jsx

```diff
@@ -19,30 +20,32 @@ const DeleteConfirmation = ({ t, selected, onConfirm, onDismiss }) => {
     title={t('deleteconfirmation.title', selected.length)}
     description={deleteConfirmationTexts}
     cancelText={t('deleteconfirmation.cancel')}
-    cancelAction={() => onDismiss(selected)}
+    cancelAction={() => onDismiss(selected, dropSelection)}
     validateType='danger'
     validateText={t('deleteconfirmation.delete')}
-    validateAction={() => onConfirm(selected)}
+    validateAction={() => onConfirm(selected, dropSelection)}
    />)
 }
 
 const mapStateToProps = (state, ownProps) => {
   return {
-    selected: state.ui.selected
+    selected: getActionableFiles(state),
+    dropSelection: state.ui.selected.length > 0
   }
 }
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
-  onDismiss: (selected) => {
+  onDismiss: (selected, dropSelection) => {
     dispatch(hideDeleteConfirmation())
-    selected.forEach(item => dispatch(toggleFileSelection(item, true)))
+    if (dropSelection) selected.forEach(item => dispatch(toggleFileSelection(item.id, true)))
   },
-  onConfirm: (selected) => {
+  onConfirm: (selected, dropSelection) => {
     selected.forEach(item => {
-      dispatch(trashFile(item))
-      dispatch(toggleFileSelection(item, true))
+      dispatch(trashFile(item.id))
+      if (dropSelection) dispatch(toggleFileSelection(item.id, true))
     })
     dispatch(hideDeleteConfirmation())
+    dispatch(hideFileActionMenu())
```

Comment:
True. As discussed with @goldoraf while coding the downloading part, we came to a conclusion that it's starting to get a little messy, although I don't think I'm the right person to clean it up 😛 
But it surely should be refactored.